### PR TITLE
tcp-port opt to expose Docker API to a local TCP address

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/docker/actions-toolkit/pull/531

Adds a `tcp-port` input to expose Docker API to a local TCP address along standard socket. Also adds `tcp` output to retrieve the full address.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -19,12 +19,20 @@ actionsToolkit.run(
...
+    }
+
     const install = new Install({
       runDir: runDir,
       source: input.source,
       rootless: input.rootless,
       contextName: input.context || 'setup-docker-action',
-      daemonConfig: input.daemonConfig
+      daemonConfig: input.daemonConfig,
+      localTCPPort: tcpPort
```
[the remark this answers is not in the document]
Comment (crazy-max) [reply src/main.ts:35]: It's used in actions toolkit: https://github.com/docker/actions-toolkit/pull/531

File: src/context.ts [hunk 2]
```diff
@@ -21,6 +22,7 @@ export function getInputs(): Inputs {
   return {
     source: source,
     daemonConfig: core.getInput('daemon-config'),
+    tcpPort: parseInt(core.getInput('tcp-port') || '0'),
```
[the remark this answers is not in the document]
Comment (crazy-max) [reply src/context.ts:25]: Ah yes would be better with `undefined`, let me change that.

File: src/main.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (vvoland) [reply src/main.ts:35]: Doh!

**Comment by vvoland** [comment]
Also, see: https://github.com/docker/actions-toolkit/pull/531#discussion_r1883801290

File: src/context.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (crazy-max) [reply src/context.ts]: Done, also opened https://github.com/docker/actions-toolkit/pull/534 to have this in our toolkit.

File: action.yml [hunk 3]
```diff
@@ -17,6 +17,10 @@ inputs:
   daemon-config:
     description: 'Docker daemon JSON configuration'
     required: false
+  tcp-port:
+    description: 'TCP port to expose the Docker API locally'
+    default: 'false'
```
Comment (vvoland) [inline action.yml:22]:
```suggestion
    default: 'false'
```

nit: Not sure what other options we have here, but is `false` a the best default for `number`?

**Review by vvoland — APPROVED** [review]

File: action.yml (hunk 3 above)
Comment (crazy-max) [reply action.yml]: Ok good catch we should not have any defaults here

Comment (vvoland) [reply action.yml]: Ah, it should be still fine, as that would still go into the `if (!value)` branch.

Comment (crazy-max) [reply action.yml]: Yes

**Pull request merged**